addons/mail/models/discuss/ir_websocket.py

```diff
@@ -11,23 +17,50 @@ class IrWebsocket(models.AbstractModel):
     _inherit = "ir.websocket"
 
     def _get_missed_presences_identity_domains(self, presence_channels):
-        identity_domain = super()._get_missed_presences_identity_domains(presence_channels)
-        if guest_ids := [
-            g.id for g, _ in presence_channels if isinstance(g, self.pool["mail.guest"])
-        ]:
-            identity_domain.append([("guest_id", "in", guest_ids)])
-        return identity_domain
+        """
+        Return a list of domains that will be combined with `expression.OR` to
+        find presences related to `presence_channels`. This is used to find
+        missed presences when subscribing to presence channels.
+
+        :param typing.List[typing.Tuple[recordset, str]] presence_channels: The
+            presence channels the user subscribed to.
+        """
+        partners = self.env["res.partner"].browse(
+            [p.id for p, _ in presence_channels if isinstance(p, self.pool["res.partner"])]
+        )
+        guest_ids = [g.id for g, _ in presence_channels if isinstance(g, self.pool["mail.guest"])]
+        # sudo: res.partner - can acess users of partner channels to find
+        # their presences as those channels were already verified during
+        # `_build_bus_channel_list`.
+        return [
+            [("user_id", "in", partners.with_context(active_test=False).sudo().user_ids.ids)],
+            [("guest_id", "in", guest_ids)]
+        ]
 
     def _get_missed_presences_bus_target(self):
         if self.env.user and not self.env.user._is_public():
-            return super()._get_missed_presences_bus_target()
+            return self.env.user.partner_id if self.env.user and not self.env.user._is_public() else None
         if guest := self.env["mail.guest"]._get_guest_from_context():
             return guest
         return None
 
     @add_guest_to_context
     def _build_presence_channel_list(self, presences):
-        channels = super()._build_presence_channel_list(presences)
+        """
+        Return the list of presences to subscribe to.
+
+        :param typing.List[typing.Tuple[str, int]] presences: The presence
+            list sent by the client where the first element is the model
+            name and the second is the record id.
+        """
+        channels = []
+        if self.env.user and self.env.user._is_internal():
+            channels.extend(
+                (partner, "presence")
+                for partner in self.env["res.partner"]
+                .with_context(active_test=False)
+                .search([("id", "in", [int(p[1]) for p in presences if p[0] == "res.partner"])])
+            )
         guest_ids = [int(p[1]) for p in presences if p[0] == "mail.guest"]
         if self.env.user and self.env.user._is_internal():
```

Comment:
I believe it should be grouped with the `is_internal` condition above. Also, the comment below is no longer accurate: `Partners already handled in super call (bus)`. I think we should unify how we register partner presence channels now that all of this can be done in the same method. Also I believe it makes sense to keep this method to avoid bloating the `_build_bus_channel_list` method.